features/netsocket/UDPSocket.h

```diff
@@ -56,6 +56,21 @@ class UDPSocket : public Socket {
      */
     virtual ~UDPSocket();
 
+    /** Subscribes to an IP multicast group
+     *
+     * @param address   Multicast group IP address
+     * @return          Negative error code on failure
+     */
+    int join_multicast_group(const SocketAddress &address);
```

Comment:
Wondering if this should be in Socket. This is falling into a trap much of the rest of the class does in that stuff ends up in UDPSocket only because it's "not TCP". Other types of socket like raw or ICMP will want this API, and it's calling an IP-level socket option.

(And Socket is currently "IP socket", rather than generic socket)

Reply:
Moved it over to Socket. Added an overload in TCPSocket to return an error for TCPSocket objects, let me know if you'd rather remove that altogether or had a better implementation.

Reply:
I don't think it is particularly important to block it on a TCP socket - it's weird but harmless. On some stacks it does work - it does make you join the group regardless. The TCP socket wouldn't process those packets itself, as it never processes multicasts, but other sockets and pings would start responding.

But if you do want to block it, it logically needs to be in setsockopt, looking at `s->conn->type` like the TCP-specific options do.